src/frequenz/client/dispatch/test/sampler.py

```diff
@@ -0,0 +1,115 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""A sampler for dispatch messages."""
+
+import random
+from datetime import datetime, timedelta, timezone
+
+from frequenz.client.common.microgrid.components import ComponentCategory
+from frequenz.client.dispatch.types import (
+    Dispatch,
+    EndCriteria,
+    Frequency,
+    RecurrenceRule,
+    Weekday,
+)
+
+
+class Sampler:
```

Comment:
If this is going to be public, I would use more clear names, like `DispatchGenerator`.